TASVideos/Extensions/ApplicationBuilderExtensions.cs

```diff
@@ -42,13 +42,29 @@ public static IApplicationBuilder UseStaticFilesWithExtensionMapping(this IAppli
 
 	public static IApplicationBuilder UseMvcWithOptions(this IApplicationBuilder app, IHostEnvironment env)
 	{
+		var permissionsPolicyValue = string.Join(", ", [
+			"camera=()", // defaults to `self`
+			"display-capture=()", // defaults to `self`
+			"fullscreen=()", // defaults to `self`
+			"geolocation=()", // defaults to `self`
+			"microphone=()", // defaults to `self`
+			"publickey-credentials-get=()", // defaults to `self`
+			"screen-wake-lock=()", // defaults to `self`
+			"web-share=()", // defaults to `self`
+
+			// ...and that's all of the non-experimental options listed on MDN as of 2024-04
+		]);
 		app.Use(async (context, next) =>
 		{
+			context.Response.Headers["Cross-Origin-Embedder-Policy"] = "credentialless";
+			context.Response.Headers["Cross-Origin-Opener-Policy"] = "same-origin";
+			context.Response.Headers["Cross-Origin-Resource-Policy"] = "cross-origin"; // TODO this is as unsecure as before; should be `same-site` or `same-origin` when serving auth-gated responses

```

Comment:
so why are we setting it to cross origin if it is "unsecure" is ther ea reason we can't do same-site or same-origin?

Reply:
Sending `CORP: same-origin` with every response will prevent JS on other sites from fetching that page. For a site whose content is mostly public, that doesn't seem fitting, but of course you may want to push third-parties towards the API rather than scraping the site. I say "unsecure", but I'm pretty sure this header only applies to the JS Fetch API, not `<iframe/>`s, and browsers won't include cookies with those requests.